### ISSUE TITLE
Extension packages with hidden option.

### ISSUE THREAD

**Issue by kdy1** [swc-project/swc#1048]
`@swc-node/jest` seems like a good idea and I think such support for some libraries worth it.

But supporting library means a maintenance hazard. What should happen if version of a library is bumped, or abandoned?
It will require bumping major version of swc, which is not a good idea.

So I think the hidden option, intended to be used only by packages in the `@swc` namespace, can be a solution.

**Comment by Brooooooklyn** [comment]
How about expose API in `@swc/core` to allow people use `native plugin` in NodeJS.
For example:

```ts
const { createTransformSync } = require('@swc/core')
const jestHoistPlugin = require('@whatever-namespace/jest')
const swcImportPlugin = require('@other-namespace/import-plugin')

const transformSync = createTransformSync(jestHoistPlugin, swcImportPlugin)

transformSync(code, swcoptions)
```

**Comment by kdy1** [comment]
I think the way you suggested is the best way to go, but I'm not sure how about the way to do it.
Long ago, I tried implementing native plugin system, but I failed because of platform issue (deployment) and neon did not support such thing.

Does napi itself supports such operation? I mean, loading other `.node`  file and invoking it?


Also, what should be done when `swc` made some breaking changes? This was one of the main problems while my experiment.
